include/cvs/common/factory.hpp

```diff
@@ -98,11 +104,11 @@ class CVSCOMMON_EXPORT Factory {
   };
 
   template <typename FactoryFunction>
-  using HelperPtr = std::shared_ptr<Helper<FactoryFunction>>;
+  using HelperUPtr = std::unique_ptr<Helper<FactoryFunction>>;
 
-  using Signatures = std::map<std::string, BaseHelperPtr>;
-  std::map<KeyType, Signatures, Compare> factory_functions;
-};
+  using Signatures = std::unordered_map<std::type_index, HelperBaseUPtr>;
+  std::unordered_map<KeyType, Signatures> factory_functions;
+};  // namespace cvs::common
```

Comment:
Это не закрытие namespace
```suggestion
};
```